### PR TITLE
Provide complete metadata for plugin type 'package-repository'.

### PR DESCRIPTION
**Pull request by varshavaradarajan**

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v2/admin/plugin_infos_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      rescue InvalidPluginTypeException
+        raise ApiV2::UnprocessableEntity, "Invalid plugins type - `#{params[:type]}` !"
+      end
+
+      def show
+        plugin = plugin_service.pluginInfo(params[:id])
+
+        raise RecordNotFound unless plugin
+
+        render json_hal_v1: ApiV2::Plugin::PluginInfoRepresenter.new(plugin).to_hash(url_builder: self)
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v2/admin/plugin_infos_controller.rb]: Use the constant in `BaseController`

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v2/admin/plugin_infos_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+##########################################################################
+
+module ApiV2
+  module Admin
+    class PluginInfosController < ApiV2::BaseController
+      before_action :check_admin_user_or_group_admin_user_and_401
+
+      def index
+        plugin_infos = plugin_service.pluginInfos(params[:type])
+        render json_hal_v1: ApiV2::Plugin::PluginInfosRepresenter.new(plugin_infos).to_hash(url_builder: self)
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v2/admin/plugin_infos_controller.rb]: Use the constant in `BaseController`

File: server/webapp/WEB-INF/rails.new/spec/presenters/api_v1/plugins/plugin_info_representer_spec.rb [hunk 3]
```diff
@@ -43,12 +43,12 @@
                                    configurations: [
                                        {
                                            key:      'url',
-                                           metadata: {:required => :true, :secure => false}
+                                           metadata: {:secure => false, :required => :true, part_of_identity: nil}
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/spec/presenters/api_v1/plugins/plugin_info_representer_spec.rb]: Use the new hash syntax please.

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v2/admin/plugin_infos_controller.rb (hunk 2 above)
Comment (varshavaradarajan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v2/admin/plugin_infos_controller.rb]: you mean `DEFAULT_FORMAT` ?

**Pull request merged**